components/Village/index.vue

```diff
@@ -108,6 +111,18 @@ export default {
     this.pagination.itemsPerPageOptions = this.generateItemsPerPageOptions
   },
   methods: {
+    searchTitle: debounce(function (value) {
+      if (value.length > 2) {
+        this.query.name = value
+      } else {
+        this.query.name = null
+        this.query.current_page = 1
+        this.query.per_page = 10
+      }
+    }, 500),
+    onSearch (value) {
+      this.searchTitle(value)
+    },
```

Comment:
Discussion:

hello, @yoslie can I ask why you choose to debounce the `query` data and not debounce the fetch function inside the watcher? Honestly, I didn't know which one is the best practice so I think it's an opportunity to learn something new here 😄

Reply:
nice discussion @Ibwedagama , did u mean like this?
honestly I got this code from @adzharamrullah 😅 

I've tried using debounce to wrap the fetch function like this:
`  watch: {
    query: {
      deep: true,
      handler: debounce(function () {
        this.$fetch()
      }, 500)
    }
  },`

and both results are similiar so I can't really answer this ✌️

Reply:
Thanks for the reply @yoslie , I try to dig into this topic and found this discussion that says:

"_This is a bit unusual, however. I don't see why you would want to debounce a watch. Possibly you would rather just debounce the`inputHandler` method._"

ref: https://stackoverflow.com/questions/45178621/how-to-correctly-use-vue-js-watch-with-lodash-debounce

But I also found an article that says you can debounce a watcher or a method
ref: https://dmitripavlutin.com/vue-debounce-throttle/

maybe both approach is fine, but I'm not entirely sure 😅